#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/Youbot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5241042879641986 0.0845746007713234 0.8474442944771717 2.9146182743743503
  position 17.934680215651877 -31.463444117899456 14.54711014007001
  follow "youBot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 0 0 -1 0.406
  floorSize 60 60
  floorTileSize 60 60
  floorAppearance PBRAppearance {
    roughness 1
    metalness 0
  }
  wallThickness 2
  wallHeight 2
}
Youbot {
  hidden position_0_0 3827.657058593214
  hidden position_0_1 6774.395701917557
  hidden position_0_2 3394.1612935091302
  hidden position_0_3 5641.432971074011
  hidden position_29_0 -5.506679713484059e-06
  hidden rotation_30 0 0 -0.9999999999999999 5.506659552487436e-06
  hidden position_30_0 0.03100421312604356
  hidden rotation_31 0 -1 0 0.031004213126045215
  hidden position_31_0 1.2275007388579326e-06
  hidden rotation_32 0 -1 0 1.2273347545846894e-06
  hidden position_32_0 -6.4597188527407605e-09
  hidden position_33_0 3.6080001299132167e-08
  hidden rotation_34 0 0 1 2.9802322387695313e-08
  hidden position_34_0 1.5688988883022983e-05
  hidden position_34_1 1.5190305365886195e-05
  hidden translation_35 0 1.5688988883022983e-05 0.09
  hidden translation_36 0 -1.5190305365886195e-05 0.09
  hidden rotation_37 0 -1 0 5.0859787863336905
  hidden rotation_38 0 1 0 1.1219407779626127
  hidden rotation_39 0 1 0 1.2412276321535485
  hidden rotation_40 0 -1 0 0.8674347732574359
  translation 10.5723 -30.5115 0.0994238
  rotation -0.0006767827057924134 -0.0012775588216285634 0.9999989549037671 3.032305774743198
  controller "youbot_controller"
  bodySlot [
    DistanceSensor {
      translation 0.308319 0.000267428 -0.03
      rotation 0.9999999999971059 -1.620879999995309e-06 -1.7779899999948542e-06 -1.5707953071795862
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_front"
      lookupTable [
        0 0 0
        1 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.308319 -0.0197325 -0.07
      rotation 0.9999999999971059 -1.620879999995309e-06 -1.7779899999948542e-06 -1.5707953071795862
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "b_front"
      lookupTable [
        0 0 0
        1 1000 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.308319 0.110267 -0.07
      rotation 0.9999999999971059 -1.620879999995309e-06 -1.7779899999948542e-06 -1.5707953071795862
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "b_front1"
      lookupTable [
        0 0 0
        1 1000 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.308319 -0.119733 -0.07
      rotation 0.9999999999971059 -1.620879999995309e-06 -1.7779899999948542e-06 -1.5707953071795862
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "b_front2"
      lookupTable [
        0 0 0
        1 1000 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.008319 0.180267 -0.03
      rotation 0.5773492691885823 0.5773502691890485 -0.5773512691895147 -2.094395307179586
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_left"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.00831892 0.180267 -0.07
      rotation 0.5773492691885823 0.5773502691890485 -0.5773512691895147 -2.094395307179586
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "b_left"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.151681 0.180267 -0.03
      rotation 0.5773492691885823 0.5773502691890485 -0.5773512691895147 -2.094395307179586
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "rear_left"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.158319 0.180267 -0.03
      rotation 0.5773492691885823 0.5773502691890485 -0.5773512691895147 -2.094395307179586
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "front_left"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.028319 -0.199733 -0.03
      rotation -0.5773512691895147 0.5773502691890485 -0.5773492691885823 2.0944
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_right"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.028319 -0.199733 -0.06
      rotation -0.5773512691895147 0.5773502691890485 -0.5773492691885823 2.0944
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "b_right"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "sonar"
    }
    Camera {
      translation 0.28 0 0.04
      children [
        Solid {
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.1
                radius 0.02
              }
            }
          ]
        }
      ]
      name "cam"
      fieldOfView 1
      width 640
      height 640
      far 10
      recognition Recognition {
        frameThickness 5
      }
    }
    DistanceSensor {
      translation 0.322588 -0.139659 -0.04
      rotation 0.1294279597735375 -0.9831056944488314 -0.1294279597735375 -1.5878353071795859
      children [
        DEF ls_right Solid {
          translation 0 0.0496411 -0.028
          rotation 1 0 0 1.0472
          children [
            DEF ls_right Shape {
              appearance PBRAppearance {
                baseColor 1 0.4 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.115
              }
            }
          ]
          name "ls_right"
        }
      ]
      name "ls_right"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.32967 0.121655 -0.0419665
      rotation 0.6232131723564657 -0.5529947563641509 -0.552993798550349 -2.0944003061004253
      children [
        DEF ls_left Solid {
          translation 0 0.0496411 -0.028
          rotation 1 0 0 1.0472
          children [
            DEF ls_left Shape {
              appearance PBRAppearance {
                baseColor 1 0.4 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.115
              }
            }
          ]
          name "ls_left"
        }
      ]
      name "ls_left"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.338319 0.000267444 -0.04
      rotation -0.44721312362157567 0.7745972141192264 0.44721312362157567 1.82348
      children [
        DEF ls_middle Solid {
          translation 0 0.0496411 -0.028
          rotation 1 0 0 1.0472
          children [
            DEF ls_middle Shape {
              appearance PBRAppearance {
                baseColor 1 0.4 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.115
              }
            }
          ]
          name "ls_middle"
        }
      ]
      name "ls_middle"
      type "infra-red"
    }
  ]
}
DEF start Solid {
  translation 47.4306 -34.9548 0
  rotation 0 0 1 3.0107
  children [
    DEF start Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "../protos/start.stl"
        ]
      }
    }
  ]
  name "start"
  recognitionColors [
    0 0 0
  ]
}
DEF end Solid {
  translation -14.8954 -19.5263 0
  rotation 0 0 1 -0.3926946938995747
  children [
    DEF end Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Mesh {
        url [
          "../protos/end.stl"
        ]
      }
    }
  ]
  name "end"
  recognitionColors [
    0 0 0
  ]
}
DEF red Solid {
  translation -15.0989 -19.442 0
  rotation 0 0 1 -0.3926946938995747
  children [
    DEF red Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0.4 0
        }
      }
      geometry Mesh {
        url [
          "../protos/letters.stl"
        ]
      }
    }
  ]
  name "LETTERS MAZE(1)"
  boundingObject USE red
  recognitionColors [
    0 0 0
  ]
}
DEF black Solid {
  translation -15.068 -19.4983 0
  rotation 0 0 1 -0.3926946938995747
  children [
    DEF black Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0.1 0
          emissiveColor 0.9 0 0
          specularColor 1 0 0
        }
      }
      geometry Mesh {
        url [
          "../protos/90Degrees.stl"
        ]
      }
    }
  ]
  name "BLACK MAZE"
  boundingObject USE black
}
DEF names Solid {
  translation -15.068 -19.4983 0
  rotation 0 0 1 -0.3926946938995747
  children [
    DEF black Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Mesh {
        url [
          "../protos/names.stl"
        ]
      }
    }
  ]
  name "BLACK MAZE(1)"
  boundingObject USE black
}
DEF maze Solid {
  translation -15.068 -19.4983 0
  rotation 0 0 1 -0.3926946938995747
  children [
    DEF black Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Mesh {
        url [
          "../protos/maze_title.stl"
        ]
      }
    }
  ]
  name "BLACK MAZE(2)"
  boundingObject USE black
}
DEF line_follower_title Solid {
  translation -15.068 -19.4983 0
  rotation 0 0 1 -0.3926946938995747
  children [
    DEF black Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Mesh {
        url [
          "../protos/line_title.stl"
        ]
      }
    }
  ]
  name "BLACK MAZE(3)"
  boundingObject USE black
}
Solid {
  translation -15.2868 -19.6795 -0.39
  rotation 0 0 -1 0.39269530610042536
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "../protos/line.stl"
        ]
      }
    }
  ]
  name "line"
}
Solid {
  translation 0.0356765 -28.4968 0.0397991
  rotation -7.04374e-17 2.94622e-17 1 -1.8400896938995748
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 1
        }
      }
      geometry Box {
        size 0.08 0.08 0.08
      }
    }
  ]
  name "BLUE BOX"
  boundingObject USE b1
  physics Physics {
  }
  recognitionColors [
    0 0 1
  ]
}
Solid {
  translation -21.0582 5.8449 0.03979909100304376
  rotation -5.249747555771988e-17 2.810179993293583e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 1
        }
      }
      geometry Box {
        size 0.08 0.08 0.08
      }
    }
  ]
  name "BLUE BOX(1)"
  boundingObject USE b1
  physics Physics {
  }
  recognitionColors [
    0 0 1
  ]
}
Solid {
  translation -19.1584 10.0794 0.03979909100304376
  rotation -7.299595048729352e-17 4.2366813447983066e-18 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.08 0.08 0.08
      }
    }
  ]
  name "BLUE BOX(2)"
  boundingObject USE b1
  physics Physics {
  }
  recognitionColors [
    1 0 0
  ]
}
Solid {
  translation 12.4391 -13.5816 -0.011275
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 1.97 3 0.04
      }
    }
  ]
  name "END"
  boundingObject USE b1
  recognitionColors [
    0 0 0
  ]
}
Solid {
  translation 11.8649 -30.6093 -0.011275
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.7091903061004252
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 1 0
        }
      }
      geometry Box {
        size 2 2 0.04
      }
    }
  ]
  name "START"
  boundingObject USE b1
  recognitionColors [
    0 0 0
  ]
}
